### PR TITLE
Introduce map polygon.

### PR DESCRIPTION
**Pull request by badnames**
## Basics

- [x] I added a line to [changelog.md](/doc/changelog.md)
- [x] The PR is rebased with current master.
- [x] Details of what you changed are in commit messages.
- [x] References to issues, e.g. `close #X`, are in the commit messages and changelog.
- [x] The buildserver is happy.

## Checklist

- [ ] I have installed and I am using [pre-commit hooks](../doc/contrib/README.md#Hooks)
- [ ] I fully described what my PR does in the documentation
- [ ] I fixed all affected documentation
- [ ] I fixed the introduction tour
- [ ] I wrote migrations in a way that they are compatible with already present data
- [ ] I fixed all affected decisions
- [ ] I added automated tests or a [manual test protocol](../doc/tests/manual/protocol.md)
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I translated all strings visible to the user
- [ ] I mentioned [every code or binary](https://github.com/ElektraInitiative/PermaplanT/blob/master/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)
- [ ] I created left-over issues for things that are still to be done
- [ ] Code is conforming to [our Architecture](/doc/architecture)
- [ ] Code is conforming to [our Guidelines](/doc/guidelines)
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)
- [ ] Exceptions to any guidelines are documented

## Review

- [ ] I've tested the code
- [ ] I've read through the whole code
- [ ] I've read through the whole documentation
- [ ] I've checked conformity to guidelines

### CODE REVIEW

**Comment by markus2330** [comment]
jenkins build please

**Comment by markus2330** [comment]
jenkins build please

**Comment by badnames** [comment]
jenkins build please

**Comment by badnames** [comment]
jenkins build please

**Review by markus2330 — CHANGES_REQUESTED** [review]
- [ ] icons at bottom
- [ ] tooltip for icons

**Comment by markus2330** [comment]
> When moving polygon Points its possible to select the entire map geometry via mouse drag.

This is not really a big problem. Would of course be nice.

- [ ] A bigger problem is that for new points the nearest point to any line should be used, not to the end-points of a line.
- [ ] Another problem is that changing between moving and adding is tiresome. It would be great if move/add can be done via middle click or with a modifier while clicking.

Maybe @danielsteinkogler can help with these two points, they are be needed for drawing layer, too.

@badnames don't forget to review #1110 :wink: Did you see the review request?

**Comment by badnames** [comment]
> Another problem is that changing between moving and adding is tiresome. It would be great if move/add can be done via middle click or with a modifier while clicking.

We should definitely avoid using middle click, as that is currently used for panning the map.
What do you think about using simple keycombinations (e.g. Ctrl.-Shift-A for adding, Ctrl.-Shift-D for deleting and Ctrl.-Shift-M for moving points) instead?

File: backend/src/service/map.rs [hunk 1]
```diff
@@ -109,3 +118,54 @@ pub async fn update(
     let result = Map::update(map_update, id, &mut conn).await?;
     Ok(result)
 }
+
+/// Update a maps gemoetry in the database.
+/// Checks if the map is owned by the requesting user.
+///
+/// # Errors
+/// If the connection to the database could not be established.
+/// If the requesting user is not the owner of the map.
```
Comment (markus2330) [inline backend/src/service/map.rs:127]:
```suggestion
/// * If the connection to the database could not be established.
/// * If the requesting user is not the owner of the map.
```

**Review by markus2330 — APPROVED** [review]
Super-nice, I was able to do the polygon for our garden in a few minutes without any annoyance. :heart: 

But there is still the problem that by default you don't see the bounding box. Please make at least the default bounding box smaller or as suggested by Yvonne (ideal solution) make the default zoom when entering a map as big as the bounding box. Can also be a follow-up.

**Comment by markus2330** [comment]
> What do you think about using simple keycombinations (e.g. Ctrl.-Shift-A for adding, Ctrl.-Shift-D for deleting and Ctrl.-Shift-M for moving points) instead?

This should be done but not instead (it is also lower-prior). The UX is not ideal due to the constant need to change between add+move. It would be better if there is only one add/move mode and when clicking on a polygon point it gets moved and when click somewhere else a new one gets created. Can also be follow-up.

Bounding box zoom fix or follow-up missing, too.

@absurd-turtle can you also take a look?

**Comment by markus2330** [comment]
I looked again how others are doing it and I didn't see a tool that combines add and move functionality, so it is probably not the best idea. What exists is that shift creates Bezier-help-points. But this is not needed for the base layer. Keybindings, however, should be added.

So pls create 2 follow-up issues, then we can hopefully merge this PR soon.

File: frontend/src/features/map_planning/layers/base/BaseLayer.tsx [hunk 2]
```diff
@@ -24,17 +27,23 @@ const BaseLayer = (props: BaseLayerProps) => {
 
   // Needed for scaling the auto-scale indicators.
   const editorLongestSide = useMapStore((map) =>
-    Math.max(map.untrackedState.editorBounds.width, map.untrackedState.editorBounds.height),
+    Math.max(map.untrackedState.editorViewRect.width, map.untrackedState.editorViewRect.height),
   );
 
+  const isBaseLayerActive =
```
[the remark this answers is not in the document]
Comment (badnames) [reply frontend/src/features/map_planning/layers/base/BaseLayer.tsx]: Thanks again for the comment. I was not aware that this util function already existed.

**Comment by badnames** [comment]
leftover issues are #1112 and #1113.

**Pull request merged**